File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        output=[]
        subset = []

        def dfs(i):
            cur_tot = sum(subset)
            if cur_tot == target:
                output.append(subset.copy())
                return
            if cur_tot > target or i >= len(nums):
                return
            
            subset.append(nums[i])
            dfs(i)
            subset.pop()
            dfs(i+1)
        
        dfs(0)

        return output

File: Data Structures & Algorithms/climbing-stairs/submission-0.java
class Solution {
    public int climbStairs(int n) {
        return dfs(n,0);
    }

    private int dfs(int target, int currentTotal){
        if (currentTotal == target){
            return 1;
        }

        if (currentTotal > target){
            return 0;
        }

        return dfs(target,currentTotal+1) + dfs(target, currentTotal+2);
    }
}
